### PR TITLE
Feature/sub events mutations

### PR DESCRIPTION
**Pull request by kevinhartmann23**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [x] Improves an existing feature
- [ ] Fixes a bug
- [x] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
*Briefly, but clearly describe what you did, and explain any new code.*
Adds mutations for subevents - add, display, update
Refactors EVENTS to render/refetch events onChange
Cleans out unused imports

## *Instructions for reviewer:*
*Where should the reviewer start?*
*How should this be manually tested?*
*Any background context you want to provide?*
*What are the relevant tickets?*

## *Issues:*
*Are there any concerns, issues, or bugs in this branch? If so describe them.*
Not at the moment, may run into merge conflicts in AppProvider

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: src/components/contacts/Contacts.js [hunk 1]
```diff
@@ -4,32 +4,6 @@ import AddContacts from './AddContacts';
...
-    lastName: 'Bair',
-    email: 'kristenbibbles@gamil.com',
-    phone: '(469) 109-8765'
-  }, {
-    id: 4,
-    firstName: 'Richard',
-    lastName: 'Tyler',
-    email: 'richardrhinosaurus@hotmail.com',
-    phone: '(897) 835-5947'
-  }];
```
Comment (DrewBradley) [inline src/components/contacts/Contacts.js:33]: Gonna miss Drew Breezy.

File: src/components/event/Event.js [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
   
+  const {loading, error} = useQuery(EVENT_BY_ID_QUERY, { 
+    variables: { id: parseInt(currentEvent.publicId)},
+  })
+
+  const eventData = client.readQuery({
+    query: EVENT_BY_ID_QUERY,
+    variables: { id: parseInt(currentEvent.publicId) }
+  })
+
```
Comment (DrewBradley) [inline src/components/event/Event.js:28]: Great job!

File: src/components/event/Event.js [hunk 3]
```diff
@@ -29,28 +42,41 @@ const Event = (props) => {
...
+  }, [])
+
+  if(loading || !eventData){
+    return <Loading />
+  }
+
+  if(error){
+    return <NotFound />
+  }
+
```
Comment (DrewBradley) [inline src/components/event/Event.js:62]: Amazing! This makes sense. Very clearly written.

File: src/components/form/Form.js [hunk 4]
```diff
@@ -78,6 +78,15 @@ const Form = ({location}) => {
...
   }
+
+  const determineSelectValue = () => {
+    const selectedId = selectedVenue === "NEW" ? venues.find(ven => ven.name === venueName).id : selectedVenue
+    setSelectedVenue(selectedId)
+  }
+
+  useEffect(() => {
+    determineSelectValue()
+  }, [venues])
```
Comment (DrewBradley) [inline src/components/form/Form.js:97]: This is a great detail. Great job, Kevin!

**Review by DrewBradley — APPROVED** [review]
You are the best!

File: src/components/event/Event.js [hunk 5]
```diff
@@ -1,18 +1,31 @@
...
 import './Event.css';
 
 const Event = (props) => {
   const [currentEvent] = useState(props.location.state.eventInfo)
-  const { deleteEvent } = useApp()
-  const [agenda, setAgenda] = useState(currentEvent.extendedProps.subEvents)
+  const { deleteEvent, setSubEventParent } = useApp()
   const history = useHistory()
   
+  const {loading, error} = useQuery(EVENT_BY_ID_QUERY, {
```
Comment (kristenmb) [inline src/components/event/Event.js:20]: This is great!

File: src/context/AppContext.js [hunk 6]
```diff
@@ -105,11 +90,21 @@ const AppProvider = ({children}) => {
...
+    onError: error => setError(error),
+  })
+
+  const [updateSubEvent] = useMutation(UPDATE_SUB_EVENT, {
     onCompleted: data => {
       console.log(data)
       setLoading(false)
       setError(false)
     },
+    refetchQueries: [{ query: EVENT_BY_ID_QUERY, variables: { id: subEventParent } }],
```
Comment (kristenmb) [inline src/context/AppContext.js:107]: This is such a cool addition

**Review by kristenmb — APPROVED** [review]

File: src/components/agenda/AgendaItem.js [hunk 7]
```diff
@@ -1,8 +1,10 @@
 import React, { useState, useEffect } from 'react';
+import {useApp} from '../../context/AppContext'
```
Comment (richardltyler) [inline src/components/agenda/AgendaItem.js:2]: global state!!!

File: src/context/AppContext.js [hunk 8]
```diff
@@ -105,11 +90,21 @@ const AppProvider = ({children}) => {
   })
 
   const [createSubEvent] = useMutation(CREATE_SUB_EVENT, {
+    onCompleted: data => {
```
Comment (richardltyler) [inline src/context/AppContext.js:93]: This `onCompleted` is sooooooo helpful

**Review by richardltyler — APPROVED** [review]
Woah theres a lot here. Thanks so much for this work kevin!

**Pull request merged**